### PR TITLE
chore: add sync pieces trace and update sync pieces logic for done task

### PR DESCRIPTION
**Pull request by jim3ma**
Signed-off-by: Jim Ma <majinjing3@gmail.com>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: client/daemon/rpcserver/rpcserver.go [hunk 1]
```diff
@@ -151,37 +152,60 @@ func (s *server) sendFirstPieceTasks(log *logger.SugaredLoggerOnWith, request *b
 func (s *server) SyncPieceTasks(sync dfdaemongrpc.Daemon_SyncPieceTasksServer) error {
 	request, err := sync.Recv()
 	if err != nil {
+		logger.Errorf("receive first sync piece tasks request error: %s", err)
```
Comment (cndoit18) [inline client/daemon/rpcserver/rpcserver.go:155]:
```suggestion
		logger.Errorf("receive first sync piece tasks request error: %s", err.Error())
```

**Review by cndoit18 — COMMENTED** [review]
之前和 @gaius-qi 的聊的看法是。
如果我们期望format一个字符串，那么可以比较明确的使用err.Error()，来表示这是一个字字符串
不然阅读起来，可能会让人联想到

```golang
fmt.Errorf("receive first sync piece tasks request error: %w", err)
```

但我们这边其实是明确一个字符串

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by gaius-qi — DISMISSED** [review]
LGTM

**Review by 244372610 — DISMISSED** [review]

**Review by 244372610 — APPROVED** [review]

**Review by yxxhero — APPROVED** [review]

**Pull request merged**